src/main/java/com/teamscale/upload/utils/FileSystemUtils.java

```diff
@@ -135,6 +137,25 @@ private static void ensureEmptyDirectory(File directory) throws IOException {
 		}
 	}
 
+	/**
+	 * Ensures that the file represents an empty directory. Creates the directory if
+	 * it doesn't exist yet.
+	 *
+	 * @throws IOException
+	 *             In case the file is not inside of the destination directory
+	 */
+	private static void ensureFileIsInsideDirectory(File file, File directory) throws IOException {
```

Comment:
Maybe rename the method to `ensureFileIsBelowDirectory`?

The javadoc says that the directory will be created if it does not exist. But I think the method will throw an IOException in that case (and never create the directory). Should we maybe split the exception into an "path does not exist" and a "is file instead of directory" exception?

Reply:
Yes this name sounds better!

The method comment is wrong, sorry for the confusion!
While unpacking the tar-archive, both files may not even exists (file never does and directory may or may not). So it should not be the responsibility of this method to check this.